src/sys/socket/addr.rs

```diff
@@ -15,12 +17,24 @@ use std::os::unix::ffi::OsStrExt;
  *
  */
 
+#[cfg(any(target_os = "linux", target_os = "android"))]
+#[derive(Copy, Clone, PartialEq, Eq, Debug, Hash)]
+#[repr(C)]
+pub struct sockaddr_nl {
```

Comment:
can we use this from libc?

http://rust-lang-nursery.github.io/libc/x86_64-unknown-linux-gnu/libc/struct.sockaddr_nl.html